transport/src/main/java/io/netty/channel/PendingWriteQueue.java

```diff
@@ -112,7 +123,6 @@ public void removeAndFailAll(Throwable cause) {
         for (PendingWrite write = head; write != null; write = head) {
             head = tail = null;
             size = 0;
```

Comment:
> should we also be setting bytes here

+1

> Should size=0 occur outside of the loop

IIUC I think this is OK. The loop basically ensures by the time this method exists there will be nothing in the queue ... it is necessary because failing promises may generate more writes. So each loop iteration the queue is emptied, and the loop doesn't terminate until the queue is empty.